### PR TITLE
Feature: use band_name when downloading albums

### PR DESCRIPTION
**Pull request by TypicalFence**
Adding a flag to organize releases by band_name instead of artist when downloading.
This is useful for when downloading stuff from labels releasing albums by many artists,

### CODE REVIEW

File: src/main/kotlin/bandcampcollectiondownloader/Args.kt [hunk 1]
```diff
@@ -36,11 +36,15 @@ data class Args(
 
         @CommandLine.Option(names = ["-e", "--skip-failed-releases"], description = ["Skip releases that fail to download after the specified number of retries."])
         var ignoreFailedReleases: Boolean = false,
+    
+        @CommandLine.Option(names = ["-b", "--use-band-name"], description = ["save Releases sorted by band name instead of artist"])
```
Comment (Ezwen) [inline src/main/kotlin/bandcampcollectiondownloader/Args.kt:40]: Here just a small fix for capital letters and the missing period would be great (`Save releases…`). Also, could we maybe have here a better description of what this distinction artist/band means in Bandcamp, for people unaware of this difference? Maybe a small sentence to explain that on Bandcamp the same artist can appear under different names in some circumstances?

**Review by Ezwen — CHANGES_REQUESTED** [review]
LGTM, apart from a small comment :+1: 

But before I merge, could you explain better to me the distinction between artist and band names? Could you give me an example of Bandcamp release (with the URL) where these values are different?

thanks in advance! cheers

File: src/main/kotlin/bandcampcollectiondownloader/Args.kt (hunk 1 above)
Comment (TypicalFence) [reply src/main/kotlin/bandcampcollectiondownloader/Args.kt:40]: consider the following 2 examples:
https://laurenbousfieldanyev3r.bandcamp.com/track/used-to-be-better
bandname: Lauren Bousfield
artist: Girls Rituals x Lauren Bousfield

https://music.businesscasual.biz/album/the-demise-of-nanami
bandname: business casual
artist: nano神社 (✪㉨✪)

I haven't doubled checked the values, out of lazyness
this feature is mostly useful when a artist has collaboration albums or it a label releasing it, where the artist name will be all over the place but you want it sorted by the label

**Comment by TypicalFence** [comment]
consider the following 2 examples:
https://laurenbousfieldanyev3r.bandcamp.com/track/used-to-be-better
bandname: Lauren Bousfield
artist: Girls Rituals x Lauren Bousfield

https://music.businesscasual.biz/album/the-demise-of-nanami
bandname: business casual
artist: nano神社 (✪㉨✪)

I haven't doubled checked the values, out of lazyness
this feature is mostly useful when a artist has collaboration albums or it a label releasing it, where the artist name will be all over the place but you want it sorted by the label

**Comment by Ezwen** [comment]
Interesting, thanks for the explanation :). I think in that case in the description of the option, we could have something like:

>"This option can be useful when an album is released on the bandcamp page of a label instead of the page of the artist, and when one wants to sort such an album per label and not per artist in the folder structure."

WDYT?

**Comment by Ezwen** [comment]
Any news?

**Comment by TypicalFence** [comment]
> Any news?

oh sorry, I was sorta busy starting a new job and sorting out my life, I'll try to make the requested changed within the next few weeks

**Comment by Ezwen** [comment]
> > Any news?
> 
> oh sorry, I was sorta busy starting a new job and sorting out my life, I'll try to make the requested changed within the next few weeks

No problem, not in a hurry, I was just curious :). Cheers.

File: src/main/kotlin/bandcampcollectiondownloader/Args.kt (hunk 1 above)
Comment (TypicalFence) [reply src/main/kotlin/bandcampcollectiondownloader/Args.kt]: Sorry this took so long, I added a sentence about the band-name

**Comment by TypicalFence** [comment]
I hope I didn't mess up anything while rebasing, it still builds tho ^^

File: src/main/kotlin/bandcampcollectiondownloader/core/BandcampCollectionDownloader.kt [hunk 2]
```diff
@@ -177,11 +179,23 @@ class BandcampCollectionDownloader(private val args: Args, private val io: IO) {
...
+
+        var releaseFolderName: String?
+        var artistFolderPath: Path?
+
+        if (!args.useBandName) {
+            artistFolderPath = downloadFolderPath.resolve(artist)
+            releaseFolderName = "$releaseYear - $releasetitle"
+        } else {
+            artistFolderPath = downloadFolderPath.resolve(bandName)
+            releaseFolderName = "$releaseYear - $artist - $releasetitle"
```
Comment (Ezwen) [inline src/main/kotlin/bandcampcollectiondownloader/core/BandcampCollectionDownloader.kt:195]: Why the `$artist` here in the folder name when `useBandName` is `false`? Any specific reason? Just being curious.

**Review by Ezwen — CHANGES_REQUESTED** [review]
Another small question for you here, other than that looks good!

File: src/main/kotlin/bandcampcollectiondownloader/core/BandcampCollectionDownloader.kt (hunk 2 above)
Comment (TypicalFence) [reply src/main/kotlin/bandcampcollectiondownloader/core/BandcampCollectionDownloader.kt:195]: seems odd I'll check it out

**Comment by Ezwen** [comment]
Small bump here as well :)

**Pull request closed without merging**